lib/primitives/address.js

```diff
@@ -404,6 +444,45 @@ class Address {
     return new this().fromBech32(data, network);
   }
 
+  /**
+   * Inject properties from bech32m address.
+   * @private
+   * @param {String} data
+   * @param {Network?} network
+   * @throws Parse error
+   */
+
+  fromBech32m(data, network) {
+    const type = Address.types.WITNESS;
+
+    assert(typeof data === 'string');
+
+    const [hrp, version, hash] = bech32m.decode(data);
+
+    assert(version > 0,
+      'Cannot convert non-program address to bech32m.');
+
+    assert(hash.length === 32,
+      'taproot program hash is the wrong size.');
```

Comment:
Ok so what we're learning from Bitcoin Core is not to do this. The reason is because in a year or so they might add a 10-byte (whatever) version 1 protocol and all bcoin users will have to update again. So we leave the length open. Anyone can make a bech32m address of any length 2 - 40 bytes. Where its important to check the length == 32 is when we are VERIFYING taproot spends.